tests/test_ping.py

```diff
@@ -0,0 +1,65 @@
+# This file is part of ctrl_bps.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Unit tests for ping.py"""
+import logging
+import unittest
+
+from lsst.ctrl.bps.ping import ping
+
+
+class TestPing(unittest.TestCase):
+    def setUp(self):
+        self.logger = logging.getLogger("test_ping")
+        self.logger.setLevel(logging.INFO)
+
+    def tearDown(self):
+        self.logger.setLevel(logging.NOTSET)
+
+    def testSuccess(self):
+        status, msg = ping("wms_test_utils.WmsServiceSuccess", None)
+        self.assertEqual(status, 0)
+        self.assertEqual(msg, "")
+
+    def testFailed(self):
+        with self.assertLogs(level=logging.INFO) as cm:
+            status, msg = ping("wms_test_utils.WmsServiceFailure", None)
+            self.assertNotEqual(status, 0)
+            self.assertEqual(msg, "Couldn't contact service X.")
+            self.assertRegex(cm.output[0], "WARN.+service failure")
+
+    def testPassThru(self):
+        with self.assertLogs(level=logging.INFO) as cm:
+            status, msg = ping("wms_test_utils.WmsServicePassThru", "EXTRA_VALUES")
+            self.assertEqual(status, 0)
+            self.assertEqual(msg, "EXTRA_VALUES")
+            self.assertRegex(cm.output[0], "INFO.+EXTRA_VALUES")
+
+    def testException(self):
+        with self.assertRaisesRegex(RuntimeError, "Service X is down"):
+            ping("wms_test_utils.WmsServiceException", None)
+
+    def testNotImplemented(self):
+        with self.assertRaises(NotImplementedError):
+            ping("wms_test_utils.WmsServiceEmpty", None)
```

Comment:
If that's the desired behavior of the `ping()` _function_ at the very least its docstring in `ping.py` should reflect that, i.e., it should have the appropriate  `Raises` section.

Reply:
Removed the testNotImplemented and testException functions.  If/when exceptions are handled by any of the code, the tests can be added then.